### PR TITLE
Android SDK New IA Improvements

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
## Pull Request Info
General cleanup:
- Clarified a potentially confusing warning
- Removed "Overview" titles from many pages where they aren't necessary
- Added links between example/fundamentals sections
- Moved data on supported field types to a specific page, since the "data types" page is impractical for the average reader to reach

### Docs staging link (requires sign-in on MongoDB Corp SSO):
Warning rewrite: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fundamentalsbacklinks/sdk/android/examples/sync-changes-between-devices/#check-the-current-network-connection
Overviews removal: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fundamentalsbacklinks/sdk/android/
Links: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fundamentalsbacklinks/sdk/android/
Field Types page: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fundamentalsbacklinks/sdk/android/data-types/field-types/

### CODE REVIEW

File: source/sdk/android/data-types/field-types.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+- ``ObjectId`` from ``org.bson.types``
+- Any ``RealmObject`` subclass
+- ``RealmList``
+
+The ``Byte``, ``Short``, ``Integer``, and ``Long`` types and their
+lowercase primitive alternatives are all stored as ``Long`` values
+within {+client-database+}. Similarly, {+client-database+} stores objects
+of the ``Float`` and ``float`` types as type ``Double``.
+
+``final`` and ``volatile`` fields are not supported, though you can
```
Comment (ccho-mongodb) [inline source/sdk/android/data-types/field-types.txt:38]: Just a suggestion that mirrors the prior paragraph and avoids lowercase at the start.

```suggestion
The ``final`` and ``volatile`` fields are not supported, though you can
```

File: source/sdk/android/examples/sync-changes-between-devices.txt [hunk 2]
```diff
@@ -187,19 +187,17 @@ To check the current network connection, call :java-sdk:`getConnectionState() <i
...
-   sync session but a network connection that is currently down.
+.. warning:: Connection States vs. Session States
+
+   The Android SDK manages communication with {+backend+} at two levels:
+   **connection state** and **session state**. *Connection state* tracks
+   the state of the network connection between a client device and your
+   backend {+app+}. *Session state* refers to a single user's
+   synchronization state, which can be :ref:`paused and resumed
+   <android-pause-or-resume-a-sync-session>` in the SDK at will. As a
+   result, you must check *both* states to come to a conclusion about
```
Comment (ccho-mongodb) [inline source/sdk/android/examples/sync-changes-between-devices.txt:198]: Nitpick: what conclusions can you arrive upon? Do you mean that you have to check both states to determined whether you are connected to the backend?

**Review by ccho-mongodb — APPROVED** [review]
LGTM! Just a couple nitpick comments.

File: source/sdk/android/data-types/field-types.txt (hunk 1 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/data-types/field-types.txt:38]: Thanks for the suggestion! I actually ended up rewording the entire sentence.

File: source/sdk/android/examples/sync-changes-between-devices.txt (hunk 2 above)
Comment (nathan-contino-mongo) [reply source/sdk/android/examples/sync-changes-between-devices.txt]: I just *knew* that my cop-out here was going to get called out. Thanks for doing it 😄 . Replaced this vagueness with the more specific "to determine whether a user's local changes will sync to the backend".

**Pull request merged**